src/create-slice.ts

```diff
@@ -1,5 +1,7 @@
 type SliceActions<Value> = {
-  [actionName: string]: (...args: never[]) => (prev: Value) => Value;
+  [actionName: string]: (
+    ...args: never[]
+  ) => (prev: Value) => Value | Promise<Value>;
```

Comment:
I don't think this is correct. Value should be a promise if we need async.

Would you please create a reproduction you want to support with stackblitz?

Reply:
> I don't think this is correct. Value should be a promise if we need async.
> 
> Would you please create a reproduction you want to support with stackblitz?

basically, you cannot have sync action & async action in one slice. 
Also the value has to be Promise for async action is strange... 
by the way, the official example is broken somehow https://codesandbox.io/p/sandbox/github/zustandjs/zustand-slices/tree/main/examples/02_async?file=%2Fsrc%2Fapp.tsx

Reply:
Sorry, it requires upcoming React 19. Let's wait for it.

Meanwhile, you can try this: 
https://twitter.com/dai_shi/status/1785504766254297436
https://stackblitz.com/edit/vitejs-vite-kxigsw?file=src%2FApp.tsx

Reply:
This might be close to what you imagine.
https://stackblitz.com/edit/vitejs-vite-n4touv?file=src%2FApp.tsx